app/main.py

```diff
@@ -1,2 +1,55 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.buckets = [[] for _ in range(self.capacity)]
+        self.length = 0
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length == self.threshold:
+            self.resize_buckets()
+        hashed_key = hash(key)
+        index = hashed_key % self.capacity
+        while True:
+
+            if not self.buckets[index]:
+                self.buckets[index] = [key, value, hashed_key]
+                self.length += 1
+                return
+            if key == self.buckets[index][0] and \
+                    hashed_key == self.buckets[index][2]:
+                self.buckets[index][1] = value
+                return
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+
+        while True:
+            if not self.buckets[index]:
+                raise KeyError
+            try:
```

Comment:
Do you still need this `try ... except` block? It seems to me that you implemented different solution and forgot to remove it.

Reply:
It's a good idea. Thank you!